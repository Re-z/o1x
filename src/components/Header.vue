<template>
    <!-- TODO: replace styles with tailwind -->
    <header class="flex items-center justify-between bg-gray-500 px-10 py-5">
        <div
            class="logo"
            @click="navigateToPage(ROUTES.MAIN_VIEW)"
        >
            <span class="first">O</span>
            <span class="second">1</span>
            <span class="third">x</span>
        </div>

        <div class="flex items-center">
            <div class="mr-8 flex">
                <AppButton
                    class="mr-8"
                    :class="{ 'bg-yellow-500': route.name === ROUTES.LIST_VIEW }"
                    :type="BUTTON_TYPES.SECONDARY"
                    text="List page exmpl"
                    @click="navigateToPage(ROUTES.LIST_VIEW)"
                />

                <AppButton
                    :class="{ 'bg-yellow-500': locale === 'ua' }"
                    :type="BUTTON_TYPES.SECONDARY"
                    text="UA"
                    @click="switchLang('ua')"
                />
                <span class="px-2 text-white">|</span>
                <AppButton
                    :class="{ 'bg-yellow-500': locale === 'en' }"
                    :type="BUTTON_TYPES.SECONDARY"
                    text="EN"
                    @click="switchLang('en')"
                />
            </div>
            <AppButton
                class="mr-8"
                :type="BUTTON_TYPES.SECONDARY"
                :text="$t('myProfile')"
            />
            <AppButton
                :type="BUTTON_TYPES.PRIMARY"
                :text="$t('createListing')"
            />
        </div>
    </header>
</template>

<script setup lang="ts">
    // imports
    import { ROUTES } from '@/router/routesMap'
    import { BUTTON_TYPES } from './generic/AppButton/AppButton'
    import AppButton from './generic/AppButton/AppButton.vue'
    import { useI18n } from 'vue-i18n'
    import { useRouter, useRoute } from 'vue-router'
    // router
    const router = useRouter()
    const route = useRoute()
    function navigateToPage(name: string) {
        router.push({
            name,
        })
    }

    // lang switch
    const { locale } = useI18n()
    function switchLang(lang: string) {
        locale.value = lang
    }
    // navigation
</script>

<style lang="scss" scoped>
    .logo {
        font-weight: bold;
        display: flex;
        align-items: center;
        cursor: pointer;
        width: fit-content;
        padding: 5px;
        .first {
            font-size: 20px;
        }
        .second {
            font-size: 25px;
        }
        .third {
            font-size: 18px;
        }
    }
</style>
