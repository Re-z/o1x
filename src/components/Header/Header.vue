<template>
    <!-- TODO: replace styles with tailwind -->
    <header class="flex items-center justify-between bg-gray-500 px-10 py-5">
        <div class="flex items-center">
            <div
                class="logo mr-5"
                @click="navigateToPage(ROUTES.MAIN_VIEW)"
            >
                <span class="first">O</span>
                <span class="second">1</span>
                <span class="third">x</span>
            </div>
            <span class="text-white">{{ $t('hello') }}, {{ user.userData.name }} ðŸ‘‹</span>
        </div>

        <div class="flex items-center">
            <div class="mr-8 flex">
                <AppButton
                    class="mr-8"
                    :class="{ 'bg-yellow-500': route.name === ROUTES.LIST_VIEW }"
                    :type="BUTTON_TYPES.SECONDARY"
                    text="List page exmpl"
                    @click="navigateToPage(ROUTES.LIST_VIEW)"
                />

                <AppButton
                    :class="{ 'bg-yellow-500': locale === LOCALES.UA }"
                    :type="BUTTON_TYPES.SECONDARY"
                    text="UA"
                    @click="switchLang(LOCALES.UA)"
                />
                <span class="px-2 text-white">|</span>
                <AppButton
                    :class="{ 'bg-yellow-500': locale === LOCALES.EN }"
                    :type="BUTTON_TYPES.SECONDARY"
                    text="EN"
                    @click="switchLang(LOCALES.EN)"
                />
            </div>
            <AppButton
                class="mr-8"
                :type="BUTTON_TYPES.SECONDARY"
                :text="$t('myProfile')"
            />
            <AppButton
                :type="BUTTON_TYPES.PRIMARY"
                :text="$t('createListing')"
            />
        </div>
    </header>
</template>

<script setup lang="ts">
    // imports
    import { ROUTES } from '@/router/routesMap'
    import { BUTTON_TYPES } from '@/components/generic/Button/Button'
    import AppButton from '@/components/generic/Button/Button.vue'
    import { useI18n } from 'vue-i18n'
    import { useRouter, useRoute } from 'vue-router'
    import { useUserStore } from '@/stores/userData'
    import { LOCALES } from '@/locales'

    // router
    const router = useRouter()
    const route = useRoute()
    function navigateToPage(name: string) {
        router.push({
            name,
        })
    }

    // lang switch
    const { locale } = useI18n()
    function switchLang(lang: string) {
        locale.value = lang
    }
    // user data
    const user = useUserStore()
</script>

<style lang="scss" scoped>
    //todo: delete this comment
    .logo {
        font-weight: bold;
        display: flex;
        align-items: center;
        cursor: pointer;
        width: fit-content;
        padding: 5px;
        .first {
            font-size: 20px;
        }
        .second {
            font-size: 25px;
        }
        .third {
            font-size: 18px;
        }
    }
</style>
