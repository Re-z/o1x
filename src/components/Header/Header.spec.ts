import { describe, it, expect } from 'vitest'
import { mountComponent } from '@/__tests__/setup'
import Header from '@/components/Header/Header.vue'

describe('Header.vue tests', () => {
    it('renders correctly (snapshot)', () => {
        const wrapper = mountComponent(Header)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
