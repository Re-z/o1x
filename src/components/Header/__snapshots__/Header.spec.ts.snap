// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.vue tests > renders correctly (snapshot) 1`] = `
"<!-- TODO: replace styles with tailwind -->
<header data-v-0c9847cd="" class="flex items-center justify-between bg-gray-500 px-10 py-5">
  <div data-v-0c9847cd="" class="flex items-center">
    <div data-v-0c9847cd="" class="logo mr-5"><span data-v-0c9847cd="" class="first">O</span><span data-v-0c9847cd="" class="second">1</span><span data-v-0c9847cd="" class="third">x</span></div><span data-v-0c9847cd="" class="text-white">Привіт, User name from store 👋</span>
  </div>
  <div data-v-0c9847cd="" class="flex items-center">
    <div data-v-0c9847cd="" class="mr-8 flex">
      <app-button-stub data-v-0c9847cd="" text="List page exmpl" type="secondary" class="mr-8"></app-button-stub>
      <app-button-stub data-v-0c9847cd="" text="UA" type="secondary" class="bg-yellow-500"></app-button-stub><span data-v-0c9847cd="" class="px-2 text-white">|</span>
      <app-button-stub data-v-0c9847cd="" text="EN" type="secondary" class=""></app-button-stub>
    </div>
    <app-button-stub data-v-0c9847cd="" text="Мій профіль" type="secondary" class="mr-8"></app-button-stub>
    <app-button-stub data-v-0c9847cd="" text="Додати оголошення" type="primary"></app-button-stub>
  </div>
</header>"
`;
